libr/include/r_util/r_event.h

```diff
@@ -13,8 +13,17 @@ typedef struct r_event_t {
 	void *user;
 	bool incall;
 	HtUP *callbacks;
+	int hook_handle_next;
```

Comment:
whats the purpose of this variable?

Reply:
unsure about this too...

Reply:
It's counted up to give every registered callback a unique handle, so you can unhook it again. Unhooking only by the function pointer will not work anymore because the same function may be hooked with a different user pointer.

Reply:
AFAIK RIDStorage should be the right thing for these kind of operations, but... I've never used it.

Reply:
Or it's RIDPool :D i'm not sure

Reply:
RIDPool would work, but looks like overkill to me.

Reply:
however, i feel like `next_handle` or something like that would be better.